src/PageContent/AssignVolunteers/AssignVolunteersController.js

```diff
@@ -59,17 +60,22 @@ class AssignVolunteersController extends Component {
         let processTimestampIndex = (timestampIndex) => {
             for (let deliveryId of Object.keys(timestampIndex)) {
                 if (!this.state.deliveries[deliveryId]) {
-                    if (!this.state.deliveriesExist) {
-                        this.setState({deliveriesExist: true});
-                    }
                     genDeliveryListener(deliveryId);
                 }
             }
         };
+
+        let updateLoading = (children, myDeliveriesRef) => {
+            if(children) { this.setState({deliveriesExist : true}); }
+            this.setState({finishedCall : true});
+        };
+
         myDeliveriesRef.on('child_added', async (snap) => processTimestampIndex(snap.val()));
```

Comment:
Originally `child_added` was sweet because it would get all existing deliveries AND update us as new ones came. However, you are totally right that we have to use `value` instead, because only then will we know if there are no items. Unfortunately, it's pretty redundant to query in `value` and also in `child_added` when we actually do have deliveries (it pull them down twice). Ideally, we would handle `genDeliveryListener` for all existing deliveries in the `value` callback, then only trigger `child_added` when _new_ deliveries are added, not for the existing ones that were already processed with `value`.

I think we might be able to get around this by only listening for children added after the current time, for example:
```
myDeliveriesRef.orderByKey().startAt(Date.now()).on('child_added', async (snap) => processTimestampIndex(snap.val());
```